packages/synapse-react-client/src/components/EntityFinder/tree/EntityTree.tsx

```diff
@@ -206,22 +206,20 @@ export function EntityTree(props: EntityTreeProps) {
     useErrorBoundary: true,
   })
 
-  const {
-    data: projectHeader,
-    isLoading: isLoadingProjectHeader,
-    error: fetchProjectError,
-  } = useGetEntityHeader(projectId!, {
-    enabled: !!(projectId ?? initialContainerPath?.path[1]?.id),
-    refetchInterval: Infinity,
-  })
+  const { data: project, isLoading: isLoadingProjectHeader } =
+    useGetEntityHeader(projectId!, undefined, {
+      enabled: !!(projectId ?? initialContainerPath?.path[1]?.id),
+      refetchInterval: Infinity,
+    })
 
   if (
     FinderScope.CURRENT_PROJECT === scope &&
     projectId &&
-    fetchProjectError &&
-    fetchProjectError.status === 403
+    !isLoadingProjectHeader &&
+    !project
   ) {
-    // The user doesn't have access to the project, so let's change the scope to something else
+    // The header wasn't returned, so the user doesn't have access to the current project
+    // Let's change the scope to something else
```

Comment:
The implementation was changed to use [POST /entity/header](https://rest-docs.synapse.org/rest/POST/entity/header.html), which is subtly different because you CAN fetch a version, but it's a batch API so it doesn't return a 403 if unauthorized.

So we have to make the assumption that if it's not present in the response body, it's because the user doesn't have access. In theory this ID came from an existing project, so assuming 403 (vs 404) seems safe.

Tracking a request for the ideal service in https://sagebionetworks.jira.com/browse/PLFM-7989